src/components/domains/EligibleDomainsSection.tsx

```diff
@@ -141,17 +89,51 @@ const UnavailableBtn = ({ domain: { owner, id } }: DomainProps) => {
 
 const DomainAction = ({ domain }: DomainProps) => {
   const { owner } = domain
-  const { promoCode } = useManageDomainContext()
-
-  let Action = BuyDomainSection
-  if (promoCode && domain.id.endsWith('.sub')) {
-    Action = ClaimFreeDomainSection
+  const { domainSellerKind } = useManageDomainContext()
+  const sellerConfig = useSelectSellerConfig()
+  const myAddress = useMyAddress()
+  const pendingOrder = useSelectPendingOrderById(domain.id)
+  const { setAddress } = useMyAccountsContext()
+  const { dmnRegPendingOrderExpTime } = sellerConfig || {}
+
+  const { createdByAccount: account, signer } = pendingOrder || {}
+
+  if (account && signer && account !== myAddress && signer !== myAddress) {
```

Comment:
Dirty :(

Maybe we can skip account && signer?

because if it undefined it is `false` here: account !== myAddress && signer !== myAddress